File: DataBaseRoomApp/jroomapp/src/main/java/com/woowrale/jroomapp/data/datasource/LocalDataSource.java
package com.woowrale.jroomapp.data.datasource;

import com.woowrale.jroomapp.domain.model.Contact;

import java.util.List;

public interface LocalDataSource {

    Boolean isEmpty();

    void saveContacts(List<Contact> contacts);

    List<Contact> getPopularContacts();

    Contact findById(Integer id);

    void update(Contact contact);
}

File: DataBaseRoomApp/kroomapp/src/main/java/com/woowrale/kroomapp/data/datasource/LocalDataSource.kt
package com.woowrale.kroomapp.data.datasource

import com.woowrale.kroomapp.domain.model.Contact

interface LocalDataSource {

    fun isEmpty(): Boolean

    fun saveContacts(contacts: List<Contact>)

    fun getPopularContacts(): List<Contact>

    fun findById(id: Int): Contact

    fun update(contact: Contact)

}